tt_metal/python_env/module.mk

```diff
@@ -6,6 +6,10 @@ python_env: $(PYTHON_ENV)/.installed
 
 python_env/dev: $(PYTHON_ENV)/.installed-dev
 
+python_env/dev/editable: $(PYTHON_ENV)/.installed-dev-editable
```

Comment:
I'm wondering if we can use this as an opportunity to quickly add `.PHONY` targets to `tt_metal/python_env/module.mk` instead of doing it manually with `touch $@` at the end. Using `.PHONY` would be more robust and we would let make handle this behaviour

Let me know if you'd like me to take care of this for you on your branch and verify that it works. It might also help with the